module/extension/Director/src/director/challenge_priority.cpp

```diff
@@ -58,23 +59,23 @@ namespace module::extension {
         }
 
         // Function to get the priorities of the ancestors of this task
-        auto get_ancestor_priorities = [this](const std::shared_ptr<const BehaviourTask>& task) {
+        auto get_ancestor_priorities = [this](const std::shared_ptr<BehaviourTask>& task) {
             // We are our first ancestor
             std::vector<TaskPriority> ancestors;
 
             // The task might be a root task, in which case we won't have any provider
             // In that case we set the type to nullptr_t to indicate that we are a root task
             ancestors.emplace_back(
-                providers.count(task->requester_id) != 0 ? providers[task->requester_id]->type : typeid(nullptr_t),
+                providers.count(task->requester_id) != 0 ? providers.at(task->requester_id)->type : typeid(nullptr_t),
```

Comment:
What's the rationale behind using `.at`? are we unconfident that the requester id is in providers? this seems odd

Reply:
In a map, if you use `operator[]` if the object doesn't exist it'll be created but when using `.at` it will just error if it doesn't exist.
For groups since there is no default constructor you can't even use `operator[]` and since I want to control when I am creating instances I use `.at`.